### PR TITLE
78 bigpop

### PR DESCRIPTION
**Pull request by jeteve**
## What does this change?

This implements the new version of the Most Popular on article, the one that's capable of holding a DMPU. See https://trello.com/c/8pXgLSzz

## Screenshots

screen shot 2018-12-07 at 10 44 41
screen shot 2018-12-07 at 10 44 29
screen shot 2018-12-07 at 10 43 46
screen shot 2018-12-07 at 10 42 49
screen shot 2018-12-07 at 10 42 10

## What is the value of this and can you measure success?

D&I have done analysis that predicts that:If we make most pop accept double MPUs in addition to singles and assume a 56% share of 300x600s with a £3 CPM, we could expect to get 70% more revenue from this slot.

### CODE REVIEW

File: common/app/views/fragments/collections/popularExtended.scala.html [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        </div>
+        }
+        @if(showMPU(containerDefinition) && isFront) {
+            <div class="fc-slice__popular-mpu fc-slice__item--mpu-candidate">
+            </div>
+        }
+        @if(isTabbed) {
+        </div>
+        </div>
+        }
```
Comment (kelvin-chappell) [inline common/app/views/fragments/collections/popularExtended.scala.html]: If you put the content of this block into a subtemplate you could avoid having the same condition twice around the opening and closing tags.
Eg. becomes something like:
```
@if(condition) {
<div>
block
</div>
} else {
block
}
```

File: common/app/views/fragments/mostPopularPlaceholder.scala.html [hunk 2]
```diff
@@ -11,23 +14,25 @@
         <div class="fc-container__inner">
             <div class="fc-container__header js-container__header">
                 <h2 class="fc-container__header__title">
-                    <a href="@LinkTo(url)" data-link-name="Most viewed @section">Most viewed</a>
+                    <a href="@LinkTo(url)" data-link-name="Most viewed @section">Most popular</a>
                 </h2>
             </div>
-            <div class="fc-container__body fc-container--rolled-up-hide popular-trails">
-                <div class="popular-trails__content js-popular-trails">
+            @defining(Switches.ExtendedMostPopular.isSwitchedOn){ isExtended =>
```
Comment (kelvin-chappell) [inline common/app/views/fragments/mostPopularPlaceholder.scala.html]: Perhaps pass this switch state into the template as a parameter?

File: common/app/views/fragments/mostPopularPlaceholder.scala.html [hunk 3]
```diff
@@ -11,23 +14,25 @@
...
+                <div class="fc-container--rolled-up-hide @if(isExtended) {most-popular__container} else {fc-container__body popular-trails}">
+                <div class="@if(isExtended) {most-popular__content} else {popular-trails__content} js-popular-trails">
                 </div>
-                <div class="popular-trails__mpu">
-                    <div class="fc-slice__popular-mpu fc-slice__item--mpu-candidate">
+                <div class="@if(isExtended) {most-popular__mpu} else {popular-trails__mpu}">
+                    <div class="@if(!isExtended) {fc-slice__popular-mpu fc-slice__item--mpu-candidate}">
```
Comment (kelvin-chappell) [inline common/app/views/fragments/mostPopularPlaceholder.scala.html]: Looks like a contradiction here

File: onward/app/feed/MostPopularAgent.scala [hunk 4]
```diff
@@ -23,25 +28,89 @@ object MostPopularRefresh {
...
+
+    val fResponse = wsClient.url(dapiURL + "/most/comments")
+      .addQueryStringParameters(params: _*)
+      .get()
+
+    fResponse.map{ r =>
+      val json = r.json
+      ( json \ "discussions" ).as[List[MostDiscussedItem]]
+        // Avoiding live blogs
+        .filter{ item => ! item.url.contains("/live/") }
```
Comment (kelvin-chappell) [inline onward/app/feed/MostPopularAgent.scala]: Could be `filterNot`.  And maybe extract function so don't need comment.
Eg. `filterNot { isLiveBlog }`

**Review by kelvin-chappell — COMMENTED** [review]
Nice! 🤑  Just a few trivial comments.

File: common/app/views/fragments/collections/popularExtended.scala.html (hunk 1 above)
Comment (jeteve) [reply common/app/views/fragments/collections/popularExtended.scala.html]: Agreed. I didn't want to risk refactoring the mostPop component though. I'll add that to the card that will say 'Get rid of the old article most pop version' once we're happy: https://trello.com/c/qETL0JDD/225-get-rid-of-the-old-most-pop-version

File: common/app/views/fragments/mostPopularPlaceholder.scala.html (hunk 2 above)
Comment (jeteve) [reply common/app/views/fragments/mostPopularPlaceholder.scala.html]: I've added that to the follow up card too: https://trello.com/c/qETL0JDD/225-get-rid-of-the-old-most-pop-version

File: onward/app/feed/MostPopularAgent.scala (hunk 4 above)
Comment (jeteve) [reply onward/app/feed/MostPopularAgent.scala]: shoulds good 👍

**Review by kelvin-chappell — APPROVED** [review]
👍

File: static/src/stylesheets/module/_most-popular.scss [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+
+.simple-content-card__headline {
+    display: block;
+
+    .fc-item__byline {
+        display: block;
+    }
+}
+
+.simple-content-card__headline .js-headline-text {
```
Comment (katebee) [inline static/src/stylesheets/module/_most-popular.scss]: With the class `.js-headline-text` we use the `js-` prefix to indicate this class is a JavaScript selector. 
Ideally, classnames should be used for styling or JS, not both.

Comment (jeteve) [reply static/src/stylesheets/module/_most-popular.scss]: I've added that in the cleanup story (https://trello.com/c/qETL0JDD/225-get-rid-of-the-old-most-pop-version) . Thanks Kate!

**Pull request merged**